### PR TITLE
Map EINPROGRESS to io::ErrorKind::WouldBlock

### PR DESCRIPTION
**Pull request by aviramha**

### CODE REVIEW

File: library/std/src/io/error.rs [hunk 1]
```diff
@@ -341,6 +341,10 @@ pub enum ErrorKind {
     #[stable(feature = "unsupported_error", since = "1.53.0")]
     Unsupported,
 
+    /// The operation was partially sucessful and needs to be checked
+    /// later on due to not blocking
+    InProgress,
```
Comment (thomcc) [inline library/std/src/io/error.rs:346]: You need an unstable annotation. Something like `#[unstable(feature = "io_error_inprogress", issue = "none")]`

**Review by thomcc — CHANGES_REQUESTED** [review]
Please read the about adding new functionality to the stdlib: https://std-dev-guide.rust-lang.org/feature-lifecycle/summary.html. In particular, this needs an API change proposal, as described here: https://std-dev-guide.rust-lang.org/feature-lifecycle/api-change-proposals.html

However, this seems redundant with [`ErrorKind::WouldBlock`](https://doc.rust-lang.org/nightly/std/io/enum.ErrorKind.html#variant.WouldBlock), so you'll need to explain how this is different and worth adding.

**Comment by aviramha** [comment]
> Please read the about adding new functionality to the stdlib: https://std-dev-guide.rust-lang.org/feature-lifecycle/summary.html. In particular, this needs an API change proposal, as described here: https://std-dev-guide.rust-lang.org/feature-lifecycle/api-change-proposals.html
> 
> However, this seems redundant with [`ErrorKind::WouldBlock`](https://doc.rust-lang.org/nightly/std/io/enum.ErrorKind.html#variant.WouldBlock), so you'll need to explain how this is different and worth adding.

Thanks for the review, followed your lead and opened an issue - https://github.com/rust-lang/libs-team/issues/92. Let's discuss there?

**Comment by thomcc** [comment]
The updated PR looks fine to me, although I'm unsure if this needs API approval. I've nominated it to sort that out.

**Comment by m-ou-se** [comment]
Closing this while we figure out what we want here: https://github.com/rust-lang/libs-team/issues/92 :)

**Pull request closed without merging**